src/analysis/normed_space/star/basic.lean

```diff
@@ -112,6 +112,45 @@ subtype.ext norm_star_mul_self
 
 end non_unital
 
+section prod_pi
+
+variables {ι R₁ R₂ : Type*} {R : ι → Type*}
+variables [non_unital_normed_ring R₁] [star_ring R₁] [cstar_ring R₁]
+variables [non_unital_normed_ring R₂] [star_ring R₂] [cstar_ring R₂]
+variables [Π i, non_unital_normed_ring (R i)] [Π i, star_ring (R i)]
+
+/-- This instance exists to short circuit type class resolution because of problems with
+inference involving Π-types. -/
+instance _root_.pi.star_ring' : star_ring (Π i, R i) := infer_instance
```

Comment:
I have to admit I'm not a fan of adding extra instances like this that solve a problem that is local to this file, but that probably doesn't do much other than add dead weight otherwise. I think I prefer the previous solution with `@`, actually. On the other hand, adding a second instance for non-dependent types seems to be fairly standard (see for instance `continuous_multilinear_map.normed_space'`), and it does help outside the file, albeit only in a special case.